### PR TITLE
Cleanup  csync_instructions_e

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/csync/csync.h [hunk 1]
```diff
@@ -103,26 +103,27 @@ Q_ENUM_NS(csync_status_codes_e)
...
-  CSYNC_INSTRUCTION_RENAME          = 0x00000004,  /* The file need to be renamed (RECONCILE) */
-  CSYNC_INSTRUCTION_EVAL_RENAME     = 0x00000800,  /* The file is new, it is the destination of a rename (UPDATE) */
-  CSYNC_INSTRUCTION_NEW             = 0x00000008,  /* The file is new compared to the db (UPDATE) */
-  CSYNC_INSTRUCTION_CONFLICT        = 0x00000010,  /* The file need to be downloaded because it is a conflict (RECONCILE) */
-  CSYNC_INSTRUCTION_IGNORE          = 0x00000020,  /* The file is ignored (UPDATE|RECONCILE) */
-  CSYNC_INSTRUCTION_SYNC            = 0x00000040,  /* The file need to be pushed to the other remote (RECONCILE) */
-  CSYNC_INSTRUCTION_STAT_ERROR      = 0x00000080,
-  CSYNC_INSTRUCTION_ERROR           = 0x00000100,
-  CSYNC_INSTRUCTION_TYPE_CHANGE     = 0x00000200,  /* Like NEW, but deletes the old entity first (RECONCILE)
+enum _csync_instructions_e {
```
Comment (ogoffart) [inline src/csync/csync.h]: SyncInsturction

**Comment by ogoffart** [comment]
Do we want that in 2.6 or should it go to master instead?
maybe safer in master. Although it is true that this probably won't break much

**Comment by TheOneRing** [comment]
Lets merge it to master

File: src/csync/csync_util.cpp [hunk 2]
```diff
@@ -40,27 +40,9 @@ Q_LOGGING_CATEGORY(lcCSyncUtils, "sync.csync.utils", QtInfoMsg)
 
 typedef struct {
   const char *instr_str;
-  enum csync_instructions_e instr_code;
+  SyncInsturctions instr_code;
 } _instr_code_struct;
```
Comment (ogoffart) [inline src/csync/csync_util.cpp]: this is no longer used.

File: src/gui/syncrunfilelog.cpp [hunk 3]
```diff
@@ -41,7 +41,7 @@ QString SyncRunFileLog::directionToStr(SyncFileItem::Direction dir)
     return re;
 }
 
-QString SyncRunFileLog::instructionToStr(csync_instructions_e inst)
+QString SyncRunFileLog::instructionToStr(SyncInsturctions inst)
```
Comment (ogoffart) [inline src/gui/syncrunfilelog.cpp]: Note that this function only consider that it has only one value,  not used as a flag.  So it should not take the QFlags

File: src/libsync/syncfileitem.h [hunk 4]
```diff
@@ -252,7 +252,7 @@ class OWNCLOUDSYNC_EXPORT SyncFileItem
     // usually this value is 1, but for removes on dirs, it might be much higher.
 
     // Variables used by the propagator
-    csync_instructions_e _instruction;
+    SyncInsturctions _instruction;
```
Comment (ogoffart) [inline src/libsync/syncfileitem.h]: As far as i know, the item can only have only one instruction at the time.
This should be a `SyncInsturction`, not the QFlags.

Comment (TheOneRing) [reply src/libsync/syncfileitem.h]: Well I can't come up with many use cases besides for tests like
`ENFORCE(instruction & (NEW | REMOVE))`
But yes maybe we should not encourage that.

**Pull request merged**